src/juliapkg/find_julia.py

```diff
@@ -103,7 +103,12 @@ def ju_find_julia(compat=None, install=False):
         return ans
     # install it
     if install:
-        ver, _ = best_julia_version(compat)
+        ver, info = best_julia_version(compat)
+        arch = info['files'][0]['arch']
+        if arch == 'aarch64':
+            ver = ver + '~aarch64'
+        elif arch == 'x64':
+            ver = ver + '~x64'
```

Comment:
I think just unconditionally do `ver += '~' + get_short_arch()` where `get_short_arch() == {'aarch64': 'aarch64', 'x86_64': 'x64', 'i686': 'x86'}[get_arch()]`. This is because (a) `info['files'][0]['arch']` will always equal `get_arch()`; and (b) `get_arch()` returns things like `x86_64` and `i686` which need translating to the short form that juliaup uses (`x64` and `x86`).

Reply:
ah make sense!